staging/src/k8s.io/apiserver/pkg/endpoints/handlers/delete.go

```diff
@@ -198,7 +200,8 @@ func DeleteCollection(r rest.CollectionDeleter, checkBody bool, scope *RequestSc
 			return
 		}
 
-		if errs := metainternalversionvalidation.ValidateListOptions(&listOptions); len(errs) > 0 {
+		metainternalversion.SetListOptionsDefaults(&listOptions, utilfeature.DefaultFeatureGate.Enabled(features.WatchList))
```

Comment:
As pointed in one of my comments above, this will eventually not be needed (once we register this function in metainternalversionscheme.ParameterCode

Maybe add a TODO to clean it up.

Same in get below.

Reply:
I will leave instructions in the comment of `SetListOptionsDefaults` function.